### ISSUE TITLE
Using .default with another control operator or a range does not work

### ISSUE THREAD

**Issue by JKRhb** [anweiss/cddl#88]
When I try to define a port number field like so:

```
example = {
    ? port: (uint .lt 65536) .default 5683
}
```

or like so

```
example = {
    ? port: 0..65535 .default 5683
}
```

I get an error when I try to validate the following JSON object:

```json
{
    "port": 5682
}
```

The error I get looks something like this:

```
error validating at cddl location "" and JSON location : CDDL member key must be string data type. got 5683
```

Using the example for `.default` and another control operator [from the specification](https://tools.ietf.org/html/rfc8610#section-3.8.6) actually also fails.

**Comment by anweiss** [comment]
Thanks for reporting @JKRhb

**Comment by anweiss** [comment]
hey @JKRhb, I'm finally getting around to addressing this. I've got a fix in place for your first example:

```
example = {
    ? port: (uint .lt 65536) .default 5683
}
```

However, your second example uses invalid syntax. The range needs to be in parenthesis like below otherwise there's ambiguity when parsing the entry.

```
example = {
    ? port: (0..65535) .default 5683
}
```

Nonetheless, the fix has been included as part of https://github.com/anweiss/cddl/pull/79

**Comment by JKRhb** [comment]
Great! Thank you for the fix and for pointing out the syntax error :)

**Comment by anweiss** [comment]
This has been fixed as of #79 and [0.9.0-beta.0](https://github.com/anweiss/cddl/releases/tag/0.9.0-beta.0). Closing
